### PR TITLE
[Tetris]: Teams Mini application template

### PR DESCRIPTION
**Pull request by abhinandan023**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "deploy-debug": "npm run build && npm run zip && upload-action-package -z ./output/ActionPackage.zip -t ./output/ -l debug --ld ./output/ --lp Tetris",
+        "update-teams-app": "npm run build && npm run zip && upload-action-package -z ./output/ActionPackage.zip --ld ./output/ --lp Tetris",
+        "inner-loop": "create-inner-loop -p <packageId> -d ./output --lp Tetris",
+        "lint": "tslint -c tslint.json -p tsconfig.json --fix"
+    },
+    "dependencies": {
+        "@fluentui/react-northstar": "0.51.0",
+        "@microsoft/m365-action-sdk": "https://static.actions.office365.com/sdk/v1.0.8/microsoft-m365-action-sdk.tgz",
+        "css-loader": "2.1.1",
+        "lodash.clonedeep": "4.5.0",
```
Comment (parthpatne) [inline package.json]: Why we are using this ?

File: src/components/Response/GameBoard.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+export default class TetrisGameBoard extends React.Component<any, any> {
+  constructor(props) {
+    super(props);
+    initialize();
+  }
+
+  render() {
+    const store = getStore();
+    let TetrisGamerows: any[] = [];
+    let copyOfTetrisBoardGrid = cloneDeep(getStore().tetrisGameBoard);
```
Comment (parthpatne) [inline src/components/Response/GameBoard.tsx]: why we are doing this cloneDeep ?

File: src/components/Response/TetrisGame.tsx [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+        let newRotate = rotate + rotateAdd > maximumRotation ? 0 : rotate + rotateAdd;
+        let rotateIsValid = true;
+
+        // Test if block should rotate
+        if (rotateAdd !== 0) {
+            for (let block = 0; block < noOfBlock; block++) {
+                // Test if block can be rotated without getting outside the board
+                if (
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][0] >= 0 &&
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][0] <= Constants.BOARD_HEIGHT &&
```
Comment (shia940) [inline src/components/Response/TetrisGame.tsx]: This looks incorrect to me. X-coordinate should be compared with Board width.

File: src/components/Response/TetrisGame.tsx [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][1] >= 0 &&
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][1] <= Constants.BOARD_HEIGHT
```
Comment (shia940) [inline src/components/Response/TetrisGame.tsx]: Looks incorrect. We should compare y-coordinate with board height. Also, blocks[activeBlockNumber][newRotate][block][1] represents y-coordinate for a tile.

File: src/components/Response/GameBoard.tsx (hunk 2 above)
Comment (abhinandan023) [reply src/components/Response/GameBoard.tsx]: cloneDeep has been resolved

File: src/components/Response/TetrisGame.tsx (hunk 3 above)
Comment (shia940) [reply src/components/Response/TetrisGame.tsx]: @abhinandan023 gentle ping for this.

File: src/components/Response/TetrisGame.tsx (hunk 4 above)
Comment (shia940) [reply src/components/Response/TetrisGame.tsx]: @abhinandan023 check this as well

File: src/components/Response/TetrisGame.tsx (hunk 3 above)
Comment (abhinandan023) [reply src/components/Response/TetrisGame.tsx]: Good catch, Now it's fixed and tested

**Pull request merged**